### PR TITLE
Setup external docker registry in serverless

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Setup external docker registry in serverless

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/8204

### CODE REVIEW

File: components/function-controller/README.md [hunk 1]
```diff
@@ -27,19 +27,18 @@ To develop the component, use the formulae declared in the [generic](/common/mak
...
-| **APP_FUNCTION_DOCKER_EXTERNAL_ADDRESS**            | External address of the Docker Registry                                                                                                                                                                                                                                                                      | `registry.kyma.local`                                                                                                                                    |
+| **APP_FUNCTION_DOCKER_INTERNAL_SERVER_ADDRESS**                     | Internal server address of the Docker Registry                                                                                                                                                                                                                                                                      | `serverless-docker-registry.kyma-system.svc.cluster.local:5000`                                                                                          |
```
Comment (kazydek) [inline components/function-controller/README.md:40]:
```suggestion
| **APP_FUNCTION_DOCKER_INTERNAL_SERVER_ADDRESS**                     | Internal server address of the Docker registry                                                                                                                                                                                                                                                                      | `serverless-docker-registry.kyma-system.svc.cluster.local:5000`                                                                                          |
```

Comment (kazydek) [reply components/function-controller/README.md:40]: Either leave `Docker Registry` and I'll change that in the whole file later on or pls change all occurrences of `Docker Registry` into `Docker registry` (lowercase). We talk about a registry in general, so I don't think we should capitalize it. Either way, let me know what you decide to do :)

File: components/function-controller/README.md [hunk 2]
```diff
@@ -27,19 +27,18 @@ To develop the component, use the formulae declared in the [generic](/common/mak
...
+| **APP_FUNCTION_DOCKER_REGISTRY_ADDRESS**            | External address of the Docker Registry                                                                                                                                                                                                                                                                      | `registry.kyma.local`                                                                                                                                    |
```
Comment (kazydek) [inline components/function-controller/README.md:41]: as above ;)

File: resources/serverless/templates/daemonset-delete-job.yaml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+{{- if .Values.dockerRegistry.enableInternal }}
```
Comment (michal-hudy) [inline resources/serverless/templates/daemonset-delete-job.yaml:1]: Hm... it should be rather removed - it was only needed for 1.14.

File: resources/serverless/templates/secrets.yaml [hunk 4]
```diff
@@ -1,28 +1,35 @@
...
+type: kubernetes.io/dockerconfigjson
+data:
+{{- if .Values.dockerRegistry.enableInternal }}
+  .dockerconfigjson: "{{- (printf "{\"auths\": {\"%s\": {\"auth\": \"%s\", \"username\":\"%s\",\"password\":\"%s\"}, \"%s\": {\"auth\": \"%s\", \"username\":\"%s\",\"password\":\"%s\"}}}" $internalServerAddress $encodedUsernamePassword $username $password $serverAddress $encodedUsernamePassword $username $password) | b64enc }}"
+{{- else }}
+  .dockerconfigjson: "{{- (printf "{\"auths\": {\"%s\": {\"auth\": \"%s\", \"username\":\"%s\",\"password\":\"%s\"}}}" $serverAddress $encodedUsernamePassword $username $password) | b64enc }}"
+{{- end }}
 metadata:
-  name: {{ template "fullname" . }}-registry-credentials
+  name: {{ template "fullname" . }}-registry-docker-config
```
Comment (michal-hudy) [inline resources/serverless/templates/secrets.yaml:16]: I don't remember right now, but will controller also cleanup secrets propagated to namespaces?

Comment (magicmatatjahu) [reply resources/serverless/templates/secrets.yaml]: Changed to older name.

File: resources/serverless/templates/daemonset-delete-job.yaml (hunk 3 above)
Comment (magicmatatjahu) [reply resources/serverless/templates/daemonset-delete-job.yaml]: I'll remove it in separate PR.

**Comment by michal-hudy** [comment]
APIRule should also be disabled when the external registry is enabled: https://github.com/kyma-project/kyma/blob/master/resources/serverless/templates/apirule.yaml

**Pull request merged**